app/models/post.rb

```diff
@@ -0,0 +1,15 @@
+class Post < ApplicationRecord
+  has_many :comments
+  has_many :likes
+  belongs_to :user
+
+  def update_counter(post_id)
+    counts = Post.where(user_id: post_id).count
+    user = User.find(post_id)
+    user.update(posts_counter: counts)
+  end
```

Comment:
- [ ] the update method should be set to private then we can use the `author.increment(:post_counter)` to update the post each time a post is added.